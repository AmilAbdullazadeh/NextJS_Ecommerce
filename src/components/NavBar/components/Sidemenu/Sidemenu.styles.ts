import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(
    ({ spacing, breakpoints, typography, palette }) => ({
        list: {
            width: spacing(30),
        },
        fullList: {
            [breakpoints.down('md')]: {
                width: 'auto',
            },
        },
        menuTitle: {
            fontSize: 18,
            fontWeight: typography.fontWeightBold,
        },
        menuHeader: {
            backgroundColor: palette.background.default,
            padding: spacing(1, 1),
        },
    })
);

export default useStyles;
