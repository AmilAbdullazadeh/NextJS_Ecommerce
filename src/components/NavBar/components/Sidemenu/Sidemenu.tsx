import React, { useState } from 'react';
import SwipeableDrawer from '@material-ui/core/SwipeableDrawer';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import LanguageIcon from '@material-ui/icons/Language';
import AccountCircleIcon from '@material-ui/icons/AccountCircle';
import ContactSupportIcon from '@material-ui/icons/ContactSupport';
import PaymentIcon from '@material-ui/icons/Payment';
import MenuIcon from '@material-ui/icons/Menu';
import IconButton from '@material-ui/core/IconButton';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';

import { Typography, Divider } from '@components';

import useStyles from './Sidemenu.styles';

const Sidemenu: React.FC = () => {
    const classes = useStyles();
    const [isOpen, setIsOpen] = useState(false);

    const openDrawer = () => setIsOpen(true);
    const closeDrawer = () => setIsOpen(false);

    const helpAndSettingsList = [
        {
            text: 'Language',
            icon: <LanguageIcon />,
        },
        {
            text: 'Account',
            icon: <AccountCircleIcon />,
        },
        {
            text: 'Payment Methods',
            icon: <PaymentIcon />,
        },
        {
            text: 'Contact Support',
            icon: <ContactSupportIcon />,
        },
    ];

    const MenuOptions: React.FC = () => (
        <div
            className={[classes.list, classes.fullList].join(' ')}
            role="presentation"
            onClick={closeDrawer}
            onKeyDown={closeDrawer}
        >
            <div className={classes.menuHeader}>
                <Typography variant="body1" color="textSecondary">
                    Hello, Sign in
                </Typography>
            </div>
            <List dense>
                <ListItem>
                    <Typography className={classes.menuTitle}>
                        Technology
                    </Typography>
                </ListItem>
                {['Smartphones', 'Laptops', 'Tablets', 'Gaming', 'Gadgets'].map(
                    text => (
                        <ListItem button key={text}>
                            <ListItemText primary={text} />
                        </ListItem>
                    )
                )}
            </List>
            <Divider />
            <List dense>
                <ListItem>
                    <Typography className={classes.menuTitle}>
                        Help & Settings
                    </Typography>
                </ListItem>
                {helpAndSettingsList.map(({ text, icon }) => (
                    <ListItem button key={text}>
                        <ListItemIcon>{icon}</ListItemIcon>
                        <ListItemText primary={text} />
                    </ListItem>
                ))}
            </List>
        </div>
    );

    return (
        <div>
            <>
                <IconButton
                    onClick={openDrawer}
                    data-testid="toggle-sidemenu-button"
                    aria-label="toggle menu"
                >
                    <MenuIcon color="secondary" />
                </IconButton>
                <SwipeableDrawer
                    anchor={'left'}
                    open={isOpen}
                    onClose={closeDrawer}
                    onOpen={openDrawer}
                >
                    <MenuOptions />
                </SwipeableDrawer>
            </>
        </div>
    );
};

export default Sidemenu;
