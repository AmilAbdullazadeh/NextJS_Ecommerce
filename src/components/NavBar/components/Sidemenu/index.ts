export { default as Sidemenu } from './Sidemenu';
