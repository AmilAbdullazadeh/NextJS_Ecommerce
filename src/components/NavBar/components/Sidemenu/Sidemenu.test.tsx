import React from 'react';

import { renderWithProviders, fireEvent, waitFor } from '@testUtils';

import Sidemenu from './Sidemenu';

describe('Sidemenu component', () => {
    it('matches snapshot', () => {
        const { container } = renderWithProviders(<Sidemenu />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('initially is hidden', () => {
        const { queryByText } = renderWithProviders(<Sidemenu />);
        const expectedText = 'Hello, Sign in';

        expect(queryByText(expectedText)).toBe(null);
    });

    it('opens the sidedrawer', async () => {
        const { queryByText, getByTestId } = renderWithProviders(<Sidemenu />);

        expect(queryByText('Hello, Sign in')).toBe(null);

        fireEvent.click(getByTestId('toggle-sidemenu-button'));

        await waitFor(() =>
            expect(queryByText('Hello, Sign in')).toBeVisible()
        );
    });

    it('closes the sidedrawer', async () => {
        const { queryByText, getByTestId, getByText } = renderWithProviders(
            <Sidemenu />
        );

        const toggleMenuBtn = getByTestId('toggle-sidemenu-button');

        toggleMenuBtn.click();

        await waitFor(() =>
            expect(queryByText('Hello, Sign in')).toBeVisible()
        );

        getByText('Hello, Sign in').click();

        await waitFor(() => expect(queryByText('Hello, Sign in')).toBe(null));
    });
});
