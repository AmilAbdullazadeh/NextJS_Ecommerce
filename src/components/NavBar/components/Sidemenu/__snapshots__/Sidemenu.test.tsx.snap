// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidemenu component matches snapshot 1`] = `
<div>
  <button
    aria-label="toggle menu"
    class="MuiButtonBase-root MuiIconButton-root"
    data-testid="toggle-sidemenu-button"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiIconButton-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
  <div
    class="PrivateSwipeArea-root-5 PrivateSwipeArea-anchorLeft-6"
    style="width: 20px;"
  />
</div>
`;
